src/client/client.ts

```diff
@@ -0,0 +1,219 @@
+      } catch (err_) {
+        err = err_;
+        // issue #675: return the raw response if the response parsing fails
+        err.rawResponse = res.text || null;
+        // issue #876: return the http status code if the response parsing fails
+        err.statusCode = res.status;
+      } finally {
+        fnOrStr(err, body);
+      }
+    });
+  };
+}
+
+/**
+ * Remove falsy values or values with a length of 0 from an object.
+ * @param obj
+ */
+function removeFalsyOrEmpty(obj: Record<string, any>) {
+  for (const key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key)) {
+      // eslint-disable-next-line no-param-reassign
+      if (!obj[key] || obj[key].length === 0) delete obj[key];
+    }
+  }
+  return obj;
+}
+
+type Query<F> = {
+  format?: F;
+  [key: string]: any;
+};
+
+/**
+ * getAcceptFormat returns the correct Accept header depending on the
+ * requested format.
+ * @param query
+ */
+/* eslint-disable no-redeclare,no-unused-vars */
+function getAcceptFormat(
+  query?: Query<'msgpack' | 'json'>
+): 'application/msgpack' | 'application/json' {
+  /* eslint-enable no-redeclare,no-unused-vars */
+  if (
+    query !== undefined &&
+    Object.prototype.hasOwnProperty.call(query, 'format')
+  ) {
+    switch (query.format) {
+      case 'msgpack':
+        return 'application/msgpack';
+      case 'json':
+      default:
+        return 'application/json';
+    }
+  } else return 'application/json';
+}
+
+export interface TokenHeader {
+  'X-Algo-API-Token': string;
+}
+
+export interface KMDTokenHeader {
+  'X-KMD-API-Token': string;
+}
+
+export default class HTTPClient {
+  private address: string;
+  private tokenHeader: TokenHeader | KMDTokenHeader;
+  public intDecoding: IntDecoding = IntDecoding.DEFAULT;
+
+  constructor(
+    tokenHeader: string | TokenHeader | KMDTokenHeader,
+    baseServer: string,
+    port?: number,
+    private defaultHeaders: Record<string, any> = {}
+  ) {
+    // Do not need colon if port is empty
+    let baseServerWithPort = baseServer;
+    if (typeof port !== 'undefined') {
+      baseServerWithPort += `:${port.toString()}`;
+    }
+    this.address = baseServerWithPort;
+    this.defaultHeaders = defaultHeaders;
+
+    // Accept token header as string or object
+    // - workaround to allow backwards compatibility for multiple headers
+    if (typeof tokenHeader === 'string') {
+      this.tokenHeader = {
+        'X-Algo-API-Token': tokenHeader,
+      };
+    } else {
+      this.tokenHeader = tokenHeader;
+    }
+  }
+
+  /**
+   * Set the default int decoding method for all JSON requests this client creates.
+   * @param {"default" | "safe" | "mixed" | "bigint"} method The method to use when parsing the
+   *   response for request. Must be one of "default", "safe", "mixed", or "bigint". See
+   *   JSONRequest.setIntDecoding for more details about what each method does.
+   */
+  setIntEncoding(method: IntDecoding) {
```

Comment:
I think the int encoding methods are no longer needed since `ServiceClient` is being used again.

Reply:
Wow, I could've sworn I removed those 🤔